pkg/kmodule/kmodule_linux_test.go

```diff
@@ -32,3 +38,162 @@ func TestGenLoadedMods(t *testing.T) {
 		}
 	}
 }
+
+// Helper function to generate compression test data for TestCompression.
+// Generates a map with the name of the file as key and the compressed data as value.
+// The data is compressed using xz, gzip, and zstd.
+// There is also one file with a bad extension to test the error handling.
+// Testing compression itself is out of scope.
+func generateCompressionTestData(data []byte) (map[string][]byte, error) {
+	var (
+		compressionBuffer bytes.Buffer
+		err               error
+	)
+
+	tData := make(map[string][]byte)
```

Comment:
This all seems very dicey, messing around with byte slices and such.
Why not just make a map[string]*bytes.Buffer

and not do all those resets and things you're doing?